### DISCUSSION TITLE
How to find out why a rule is unsafe?

### DISCUSSION THREAD

**Discussion by Spectre5** [astral-sh/ruff#12071] [General]
What is the easiest way to find out why any given fix is marked as unsafe?  It would be awesome if the rule pages for unsafe fixes gave an example of it being unsafe and/or describe why it can be unsafe.

I know the docs give a good example for one rule where the raised exception could be changed based on the fix.  Having something like that for each usafe fix would be great.

**Comment by charliermarsh** [comment]
We now add `## Fix safety` documentation to all new rules, and occasionally add it for existing rules though we haven't backfilled all of them: https://docs.astral.sh/ruff/rules/assert-false/

**Reply by Spectre5** [reply]
Got it, I wondered if that was the case as I have seen the fix safety section on some rules, just not all.  If I wanted to help add that safety section to some rules, is there an easy way to find out why it was marked unsafe (in the code or something)?

On the rules overview page of the documentation, would it be possible to include another icon, or a different one for the fix icon, when it is unsafe?

**Reply by dhruvmanila** [reply]
>  If I wanted to help add that safety section to some rules, is there an easy way to find out why it was marked unsafe (in the code or something)?

Thank you for showing interest in helping us with that. I think the best way would to (1) look at the code and (2) look at the originating PR where the rule / fix was added and marked as unsafe.
